registry/regsrc/module_test.go

```diff
@@ -119,9 +111,6 @@ func TestModule(t *testing.T) {
 				return
 			}
 
-			if v := got.String(); v != tt.wantString {
-				t.Fatalf("String() = %v, want %v", v, tt.wantString)
-			}
```

Comment:
Why remove these? The main motivation for this package was to encapsulate normalization rules but also to allow us to retain the original input from the user (i.e. the original case used etc.).

Several of these test cases were setup to exercise that property. I don't have a specific reason to require that we can reconstruct exactly however in hand-wavy terms it is the case that we have providers with capitalised GitHub org names which we present in original case everywhere on the registry itself (including currently in API responses as module names).

So I'm not sure. It probably doesn't matter and simplifies things slightly to remove this, but then I'd argue that `Display` and `String` now have a confusing distinction. Neither has the original-preserving property and `String` is the one you actually use for display not `Display`...

If we really want to simplify and don't care about preserving input format, we could probably rewrite this whole package to be about 10 lines of code :)

I'm open to that. But this seems to leave it in a strange inconsistent state unless I'm miss-understanding something...

Reply:
Yes, the Hostname implementation doesn't store the original string, only the normalized. I think we just need to merge the functionality of the two hostname types, rather than bolting them together.